zebra-utils/src/bin/zebra-checkpoints/main.rs

```diff
@@ -1,32 +1,108 @@
+//! Prints Zebra checkpoints as "height hash" output lines.
+//!
+//! Get all the blocks up to network current tip and print the ones that are
+//! checkpoints according to rules.
+//!
+//! Usage: zebra-checkpoints [network] --cli <cli-path>
+//! Where network can be `--regtest`, `mainnet` or `--testnet`.
+//! `--cli` is the path of the zcash-cli binary as a string.
+//! If no network is specified program will use the mainnet.
+//!
+//! zebra-consensus accepts an ordered list of checkpoints, starting with the
+//! genesis block. Checkpoint heights can be chosen arbitrarily.
+
 #![allow(clippy::try_err)]
-use color_eyre::eyre::{eyre, Result};
+use color_eyre::eyre::Result;
+use serde_json::Value;
+use std::process::Stdio;
 use structopt::StructOpt;
 
 mod args;
 
+/// We limit the maximum number of blocks in each checkpoint. Each block uses a
+/// constant amount of memory for the supporting data structures and futures.
+const MAX_CHECKPOINT_HEIGHT_GAP: usize = zebrad::commands::LOOKAHEAD_LIMIT / 2;
+
+/// We limit the memory usage for each checkpoint, based on the cumulative size of
+/// the serialized blocks in the chain. Deserialized blocks are larger, because
+/// they contain pointers and non-compact integers. But they should be within a
+/// constant factor of the serialized size.
+const MAX_CHECKPOINT_BYTE_COUNT: i64 = 256 * 1024 * 1024;
+
+/// Checkpoints must be on the main chain, so we skip blocks that are within the
+/// zcashd reorg limit.
+const BLOCK_REORG_LIMIT: i64 = 100;
```

Comment:
BlockHeight is a u32 internally, but we should just use the type here:
```suggestion
const BLOCK_REORG_LIMIT: BlockHeight = 100;
```